lib/client.js

```diff
@@ -1,60 +1,108 @@
+/* eslint-disable no-await-in-loop */
+const retry = require('async-retry');
 const { google } = require('googleapis');
+const utils = require('../lib/helpers/utils');
+const { getRetriesFromConfig } = require('./common');
+
+let token;
+let secret;
 
 class GoogleOauth2Client {
   constructor(configuration, context) {
     if (!context || !context.logger || !context.emit) {
       throw new Error('Can not find a valid context.');
     }
-
     this.configuration = configuration;
-    this.context = context;
+    this.logger = context.logger;
+  }
 
-    this.getToken();
-    this.getCredentials();
-    this.buildClient();
+  // eslint-disable-next-line class-methods-use-this
+  resetSecret() {
+    secret = null;
   }
 
-  getToken() {
-    const { oauth } = this.configuration;
-    if (!this.token) {
-      this.token = {
-        access_token: oauth.access_token,
-        refresh_token: oauth.refresh_token,
-        expires_in: oauth.expires_in,
-        scope: oauth.scope,
-        token_type: oauth.token_type || 'Bearer',
-      };
+  async getToken() {
+    const { secretId } = this.configuration;
+    if (!secret) {
+      if (!secretId) {
+        secret = {
+          credentials: this.configuration.oauth.oauth2.keys,
+        };
+      } else {
+        this.logger.info('Going to load the new secret...');
+        secret = await utils.getSecret(this, secretId);
+      }
+    } else {
+      this.logger.info('Secret cached: %j', secret);
     }
-    return this.token;
+    const { credentials } = secret;
+    this.logger.info('Secret credentials: %j', credentials);
```

Comment:
IIRC there is only a secretId in the secret? No sensitive data?
Probably should be on the 'debug' level